torch/lib/c10d/ProcessGroup.hpp

```diff
@@ -297,6 +324,8 @@ class ProcessGroup : public torch::CustomClassHolder {
  protected:
   const int rank_;
   const int size_;
+  // Optional sequence number structure for matching collectives.
+  c10::optional<c10d::SequenceNum> sequenceNum_ = c10::nullopt;
```

Comment:
Nit: shall we prefer std::optional to c10d::optional?

Reply:
I think std::optional is only available in C++17 (https://en.cppreference.com/w/cpp/utility/optional) which pytorch OSS doesn't support yet afaik. The only uses of `std::optional` in `torch/` directory seems to be in fb-only directories.